lib/pos_blockchain.go

```diff
@@ -570,17 +570,25 @@ func (bc *Blockchain) validateLeaderAndQC(block *MsgDeSoBlock) (_passedSpamPreve
 	if !isValidPartialSig {
 		return false, nil
 	}
-	// 2. Validate QC
-	validatorsByStake, err := utxoView.GetAllSnapshotValidatorSetEntriesByStake()
+
+	// 2. Fetch the validator set for the current and previous block heights.
+	validatorSetByBlockHeight, err := utxoView.GetSnapshotValidatorSetsForBlockHeights([]uint64{
+		block.Header.Height, block.Header.Height - 1,
```

Comment:
Do we want some underflow check? I mean it won't ever really happen I guess.

Reply:
ProcessBlockPoS verifies that the incoming block has a high after the PoS cutover block height. It's sufficient to rely on that since ProcessBlockPoS is not compatible with 0 height blocks anyway